k8s-operator/apis/v1alpha1/types_connector.go

```diff
@@ -77,6 +92,40 @@ type SubnetRouter struct {
 	Hostname Hostname `json:"hostname,omitempty"`
 }
 
+type Tags []Tag
+
+func (tags Tags) Stringify() []string {
+	stringTags := make([]string, len(tags))
+	for i, t := range tags {
+		stringTags[i] = string(t)
+	}
+	return stringTags
+}
+
+// Exit node describes a Tailscale exit node.
+// +kubebuilder:validation:XValidation:rule="has(self.tags) == has(oldSelf.tags)",message="Exit node tags currently cannot be changed. Delete and redeploy the Connector if you need to change them."
+type ExitNode struct {
+	// Hostname is the tailnet hostname that should be assigned to the exit
+	// node. If unset hostname is defaulted to <connector name>-exitnode.
+	// Hostname can contain lower case letters, numbers and dashes. It must
+	// not start or end with a dash and must be between 2 and 63 characters
+	// long.
+	// +optional
+	Hostname Hostname `json:"hostname,omitempty"`
+	// Tags are Tailscale ACL tags that the exit node will be tagged with.
+	// If you want the exit node to be autoapproved, you can configure
+	// Tailscale ACLs to autoapprove exit nodes for these
+	// tags. See
+	// https://tailscale.com/kb/1018/acls/#auto-approvers-for-routes-and-exit-nodes
+	// Tags defaults to tag:k8s. If you specify custom tags here, you must
+	// also make tag:k8s-operator owner of the custom tag. See
+	// https://tailscale.com/kb/1236/kubernetes-operator/#setting-up-the-kubernetes-operator.
+	// Tags cannot be changed once a Connector has been created. Tag values
+	// must be in form ^tag:[a-zA-Z][a-zA-Z0-9-]*$.
+	// +optional
+	Tags Tags `json:"tags,omitempty"`
```

Comment:
Arguably it tags and hostname could have been shared top level fields in Connector spec. But having them defined for each allows for these component-specific comments that describe the component-specific defaults.

Reply:
(Out of date comment: it is now at top level)